docs/core/diagnostics/metrics-collection.md

```diff
@@ -7,97 +7,83 @@ ms.date: 10/27/2021
-some trivial metric instrumentation. The .NET runtime also has [various metrics built-in](available-counters.md). For more information about creating new metrics using the
-<xref:System.Diagnostics.Metrics.Meter?displayProperty=nameWithType> API shown here, see
-[the instrumentation tutorial](metrics-instrumentation.md).
+## Create an example app
+
+Before metrics can be collected, measurements must be produced. This tutorial creates an app that has basic metric instrumentation. The .NET runtime also has [various metrics built-in](available-counters.md). For more information about creating new metrics using the <xref:System.Diagnostics.Metrics.Meter?displayProperty=nameWithType> API, see [the instrumentation tutorial](metrics-instrumentation.md).
 
 ```dotnetcli
 dotnet new console
 dotnet add package System.Diagnostics.DiagnosticSource
 ```
 
-Replace the code of `Program.cs` with:
+Replace the contents of `Program.cs` with the following code:
 
-```csharp
-using System;
-using System.Diagnostics.Metrics;
-using System.Threading;
+:::code language="csharp" source="snippets/Metrics/Program.cs" id="snippet":::
 
-class Program
-{
-    static Meter s_meter = new Meter("HatCo.HatStore", "1.0.0");
-    static Counter<int> s_hatsSold = s_meter.CreateCounter<int>("hats-sold");
-
-    static void Main(string[] args)
-    {
-        Console.WriteLine("Press any key to exit");
-        while(!Console.KeyAvailable)
-        {
-            // Pretend our store has a transaction each second that sells 4 hats
-            Thread.Sleep(1000);
-            s_hatsSold.Add(4);
-        }
-    }
-}
-```
+The preceding code simulates selling hats at random intervals and random times.
 
 ## View metrics with dotnet-counters
 
-[dotnet-counters](dotnet-counters.md) is a simple command-line tool that can view live metrics for any .NET Core application on
-demand. It doesn't require any advance setup, which can make it useful for ad-hoc investigations or to verify that metric
-instrumentation is working correctly. It works with both <xref:System.Diagnostics.Metrics?displayProperty=nameWithType> based
-APIs and [EventCounters](event-counters.md).
+[dotnet-counters](dotnet-counters.md) is a command-line tool that can view live metrics for .NET Core apps on demand. It doesn't require setup, making it useful for ad-hoc investigations or verifying that metric instrumentation is working. It works with both <xref:System.Diagnostics.Metrics?displayProperty=nameWithType> based APIs and [EventCounters](event-counters.md).
+
+If the [dotnet-counters](dotnet-counters.md) tool isn't installed, run the following command:
+
+```dotnetcli
+dotnet tool update -g dotnet-counters
+```
 
-If the [dotnet-counters](dotnet-counters.md) tool is not already installed, use the SDK to install it:
+While the example app is running, list the running processes in another console to determine the process ID:
```

Comment:
We can shorten these steps now that dotnet-counters supports attach by process name using the -n [switch](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-counters#options-1). I'm not sure how to nest more markup inside of a suggestion, but hopefully this conveys the right idea:

```suggestion
While the example app is running, launch [dotnet-counters](dotnet-counters.md). The following command shows an example of `dotnet-counters` monitoring all metrics from the `HatCo.HatStore` meter. The meter name is case-sensitive. Our sample app was metric-instr.exe, substitute this with the name of your sample app.

dotnet-counters monitor -n metric-instr --counters Hatco.HatStore
```

This one step replaces the previous three steps of dotnet-counters ps, searching the output to find the process ID, and then using that process ID in the dotnet-counters monitor command.

If this change is best done as a separate PR feel free to leave it out.